### ISSUE TITLE
Is there a way to enable/disable features when adding tenants?

### ISSUE THREAD

**Issue by MikeAlhayek** [OrchardCMS/OrchardCore#7775]
I am trying to use the SaaS recipe to create a multi-tenant application. Currently, it seems that I have to create a predefined receipt in order to create a tenant which is fine is most cases. But what if I want to use the recipe as the starting guide then add/remove some features as needed?

I think it may be very useful to allow the user to check/uncheck features from the control panel directly. For example, after a user selects a recipe, show all the feature that will be included with an option to uncheck default feature. At the same time, give the user option to add features all from the "Create Tenant" screen. Then when a tenant is created, use the **selected** instead of the predefined features to create the new tenant. 

Additionally, I want to be able to add/remove features using the edit tenant screen. In my case, I don't want to the site owner to have access to the Features module, instead, I want to add/remove features for my tenants. Having this feature, is important when you want to control what a tenant can have/do based on the agreed contract with them.

Thanks for your consideration.

**Comment by Skrypt** [comment]
When you add a tenant through DI you can specify which assemblies you want to include in your tenant. These modules needs to be referencing OrchardCore.Module.Target simply.

**Comment by Skrypt** [comment]
And I've read back your post there and this is a feature that has been asked before. Restrict modules per tenant.

**Comment by CrestApps** [comment]
@Skrypt, I feel that the function is there for the most part. I think it is only missing the ability to lock/unlock features using checkbox option in the UI. The add screen would show a list of features that are set to be unlocked by default, then the user is able to manually check/uncheck items from that list as they need. Similarly, lock/unlock feature from the edit screen.

**Comment by sebastienros** [comment]
I think I agree with the request:
- manage tenant features from the default tenant (from the tenants feature)
- be able to enable/disable a feature for a set or all non-default tenants
- be able to run a recipe (text area) for a set or all non-default tenants

But not from the tenant setup screen which has to remain focused.

**Comment by CrestApps** [comment]
@sebastienros I agree with you that the setup screen should remain focused. However, I don't think it eliminate the focus if under the recipe field we add a list of a checkboxes that shows what features will be selected based on this selected recipe. Then if one selected the "Blog" recipe for example, but then check one more additional feature. If you think about it, if you want a new recipe which enables blog + 1 more module, or a blog recipe with 2 more modules, it does not make scene to have to create a new recipe for every combination possible. If the admin "whom is admin a new site" want to only run a specific recipe, he/she won't modify the checkbox which will be check by default after the recipe is selected.

Something like this is what I am talking about. The recipe will check the features based on the recipe, but then the user can change the recipe before it is executed.

I hope this make more since and become acceptable.
![image](https://user-images.githubusercontent.com/24724371/102537277-9d4be680-405f-11eb-94f4-ec20b5165f23.png)

**Comment by CrestApps** [comment]
@sebastienros I agree with you that the setup screen should remain focused. However, I don't think it eliminate the focus if under the recipe field we add a list of a checkboxes that shows what features will be selected based on this selected recipe. Then if one selected the "Blog" recipe for example, but then check one more additional feature. If you think about it, if you want a new recipe which enables blog + 1 more module, or a blog recipe with 2 more modules, it does not make scene to have to create a new recipe for every combination possible. If the admin "whom is admin a new site" want to only run a specific recipe, he/she won't modify the checkbox which will be check by default after the recipe is selected.

Something like this is what I am talking about. The recipe will check the features based on the recipe, but then the user can change the recipe before it is executed.

I hope this make more since and become acceptable.
![image](https://user-images.githubusercontent.com/24724371/102537277-9d4be680-405f-11eb-94f4-ec20b5165f23.png)
